### PR TITLE
feat: validation report

### PR DESCRIPTION
**Pull request by jujiang526**
**Summary:**

add validation report for gtfs datasets

**Expected behavior:** 

Explain and/or show screenshots for how you expect the pull request to work in your testing (in case other devices exhibit different behavior).

**Testing tips:**

Provide tips, procedures and sample files on how to test the feature.
Testers are invited to follow the tips AND to try anything they deem relevant outside the bounds of the testing tips. 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [ ] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: api/src/feeds/impl/datasets_api_impl.py [hunk 1]
```diff
@@ -39,6 +47,38 @@ def create_dataset_query():
             .join(Feed, Feed.id == Gtfsdataset.feed_id)
         )
 
+    @staticmethod
+    def _load_validation_report():
```
Comment (davidgamez) [inline api/src/feeds/impl/datasets_api_impl.py:51]: Let's add unit tests that cover this function.

File: api/src/feeds/impl/datasets_api_impl.py [hunk 2]
```diff
@@ -95,10 +135,34 @@ def get_datasets_gtfs(query: Query, limit: int = None, offset: int = None) -> Li
...
+        notices = DatasetsApiImpl._load_validation_report()
+
         gtfs_datasets = []
         for dataset_group in dataset_groups:
             dataset_objects, components, bound_box_strings, feed_ids = zip(*dataset_group)
             database_gtfs_dataset = dataset_objects[0]
+            notices_for_dataset = [notice for notice in notices if notice.dataset_id == database_gtfs_dataset.id]
+
+            validator_report = None
+            if notices_for_dataset:
```
Comment (davidgamez) [inline api/src/feeds/impl/datasets_api_impl.py:147]: Let's add unit tests that cover this block of code.

Comment (jujiang526) [reply api/src/feeds/impl/datasets_api_impl.py:150]: added unit tests for validation report

File: api/src/feeds/impl/datasets_api_impl.py (hunk 1 above)
Comment (jujiang526) [reply api/src/feeds/impl/datasets_api_impl.py:54]: this method is covered by the unit tests for the validation report

**Comment by jujiang526** [comment]
@davidgamez merge conflicts should have been resolved

**Review by davidgamez — APPROVED** [review]
LGTM!

**Pull request merged**